pkg/scheduler/queue/tenant_queues.go

```diff
@@ -140,11 +147,20 @@ func (qb *queueBroker) enqueueRequestFront(request *tenantRequest, tenantMaxQuer
 
 func (qb *queueBroker) makeQueuePath(request *tenantRequest) (QueuePath, error) {
 	if qb.additionalQueueDimensionsEnabled {
+		dimensions := QueuePath{}
 		if schedulerRequest, ok := request.req.(*SchedulerRequest); ok {
+			// TODO (casie): Clean this up when deprecating old TreeQueue
+			if _, ok := qb.tree.(*MultiQueuingAlgorithmTreeQueue); ok && len(schedulerRequest.AdditionalQueueDimensions) == 0 {
+				// if using new MultiQueueingAlgorithmTreeQueue and AdditionalQueueDimensions is empty,
+				// enqueue request to a node "unknown" to keep all requests enqueued on leaf nodes.
+				dimensions = append(dimensions, unknownQueueDimension)
+			} else {
+				dimensions = schedulerRequest.AdditionalQueueDimensions
+			}
```

Comment:
let's leverage `schedulerRequest.ExpectedQueryComponentName()` for this if we can? It's supposed to help hide the details of grokking this stuff from `AdditionalQueueDimensions` which will probably go away/move/be renamed.